### PR TITLE
Update Aftershock and Magiclysm as much as possible to work with Brig…

### PR DESCRIPTION
**Pull request by Kenan2000**
Bright Nights

#### Summary
SUMMARY: **Mods** "Updates Aftershock and Magiclysm to work properly on Bright Nights along with new content additions"

#### Purpose of change

To add new content to Aftershock and Magiclysm Bright Nights versions

#### Describe the solution
Adds latest new content from Aftershock and Magiclysm relying on all the Bright Nights mechanics without any incompatibilities or breaking anything

#### Describe alternatives you've considered
Do not include new content updates

#### Testing
Tested thoroughly both Aftershock and Magiclysm, tested the entire debug spawn item list along with debug monster lists - no errors, everything works as intended, no non-existent items or spells, no errors of whatsoever, everything works perfectly fine and nicely, loads without any errors of whatsoever successfully 

#### Additional context
None

### CODE REVIEW

File: data/mods/Aftershock/field.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[
+  {
+    "id": "fd_clairvoyant",
```
Comment (Coolthulhu) [inline data/mods/Aftershock/field.json:3]: Clairvoyance is not supported in BN (it's too expensive to compute).

File: data/mods/Aftershock/items/ammo_effects.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "id": "FOAMCRETE",
+    "type": "ammo_effect",
+    "foamcrete_build": true
```
Comment (Coolthulhu) [inline data/mods/Aftershock/items/ammo_effects.json:5]: Did you test if this works? It doesn't seem to reference any implemented field.

File: data/mods/Aftershock/items/comestibles/cheap_food.json [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    "calories": 2000,
+    "healthy": 3,
+    "quench": -5,
+    "description": "A crumbly white pill half the size of your thumb, packed with vitamins and calories.  Provides a whole day's nutrition in portable form, but it's a little hard to keep down, thanks to the rancid taste and chalky texture.",
+    "price": 10000,
+    "volume": "5 ml",
+    "charges": 5,
+    "looks_like": "vitamins",
+    "fun": -8,
+    "freezing_point": -9999,
```
Comment (Coolthulhu) [inline data/mods/Aftershock/items/comestibles/cheap_food.json:19]: Obsolete field

File: data/mods/Aftershock/items/comestibles/comestibles.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    "comestible_type": "MED",
+    "name": { "str": "Panaceus", "str_pl": "Panaceii" },
+    "symbol": "!",
+    "color": "red",
+    "copy-from": "panacea",
+    "use_action": {
+      "type": "consume_drug",
+      "activation_message": "You feel AMAZING!",
+      "effects": [ { "id": "panacea", "duration": "1 m" }, { "id": "cureall" } ],
+      "damage_over_time": [
```
Comment (Coolthulhu) [inline data/mods/Aftershock/items/comestibles/comestibles.json:34]: Not implemented

File: data/mods/Aftershock/items/materials.json [hunk 5]
```diff
@@ -6,6 +6,7 @@
     "density": 10,
     "bash_resist": 4,
     "cut_resist": 9,
+    "bullet_resist": 7,
```
Comment (Coolthulhu) [inline data/mods/Aftershock/items/materials.json:9]: Not implemented

File: data/mods/Aftershock/field.json (hunk 1 above)
Comment (Kenan2000) [reply data/mods/Aftershock/field.json:3]: Will handle these shorty

File: data/mods/Aftershock/items/obsolete.json [hunk 6]
```diff
@@ -32,7 +80,8 @@
     "covers": [ "TORSO", "ARMS", "HANDS", "LEGS", "FEET" ],
     "coverage": 100,
     "material_thickness": 2,
-    "encumbrance": 9,
+    "encumbrance": 4,
+    "max_encumbrance": 9,
```
Comment (Coolthulhu) [inline data/mods/Aftershock/items/obsolete.json:84]: Not implemented

File: data/mods/Aftershock/items/tool_armor.json [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        "battery",
+        [
+          "medium_battery_cell",
+          "medium_plus_battery_cell",
+          "medium_atomic_battery_cell",
+          "medium_disposable_cell"
+        ]
+      ]
+    ],
+	"rigid": true,
```
Comment (Coolthulhu) [inline data/mods/Aftershock/items/tool_armor.json:29]: Tabs

File: data/mods/Aftershock/items/tools.json [hunk 8]
```diff
@@ -104,7 +104,7 @@
     "looks_like": "pot",
     "symbol": ")",
     "color": "light_green",
-    "use_action": "cauterize",
+    "use_action": [ "HOTPLATE" ],
```
Comment (Coolthulhu) [inline data/mods/Aftershock/items/tools.json:107]: Disabled. Use "cauterize".

File: data/mods/Aftershock/mobs/PrepPhyle_mobs.json [hunk 9]
```diff
@@ -56,6 +57,7 @@
     "harvest": "mammal_tiny",
     "path_settings": { "max_dist": 10 },
     "upgrades": { "age_grow": 42, "into": "afs_mon_cyber_mastiff" },
+    "biosignature": { "biosig_item": "feces_dog", "biosig_timer": 6 },
```
Comment (Coolthulhu) [inline data/mods/Aftershock/mobs/PrepPhyle_mobs.json:60]: Not supported

File: data/mods/Aftershock/mobs/uplifted_monsters.json [hunk 10]
```diff
@@ -0,0 +1,113 @@
+[
+  {
+    "id": "mon_uplifted_bear",
+    "type": "MONSTER",
+    "name": { "str": "Schwarz Walder" },
+    "description": "Schwarz Walders were originally developed by a German company as forest rangers for the expanded Black Forest Conservation Area.  Shortly afterwards it was determined that they also made excellent long range reconnaissance units that could operate independently indefinitely.  Pre-Cataclysm they lived on every continent in a variety of jobs.",
+    "default_faction": "UPLIFT",
+    "bodytype": "bear",
```
Comment (Coolthulhu) [inline data/mods/Aftershock/mobs/uplifted_monsters.json:8]: Not supported

File: data/mods/Aftershock/mobs/uplifted_monsters.json [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+        "targeting_volume": 20,
+        "no_ammo_sound": "a chk!"
+      }
+    ],
+    "vision_day": 30,
+    "vision_night": 10,
+    "path_settings": { "max_dist": 10 },
+    "anger_triggers": [ "HURT", "PLAYER_NEAR_BABY" ],
+    "death_function": [ "NORMAL" ],
+    "zombify_into": "mon_zombie_upliftedbear",
```
Comment (Coolthulhu) [inline data/mods/Aftershock/mobs/uplifted_monsters.json:52]: Not supported

File: data/mods/Aftershock/mobs/uplifted_monsters.json (hunk 10 above)
Comment (Kenan2000) [reply data/mods/Aftershock/mobs/uplifted_monsters.json:8]: Can you elaborate which one of these is not supported?I'm having a difficulty here and in uplifted_monsters.json file in general 😕

Comment (Kenan2000) [reply data/mods/Aftershock/mobs/uplifted_monsters.json:8]: Ok, figured out the other one but cannot figure out this one - I've searched and this JSON seems to be supported in vanilla Bright Nights, both "bodytype", "default_faction", so even more confused, what exactly is not supported here 😕

Comment (Coolthulhu) [reply data/mods/Aftershock/mobs/uplifted_monsters.json:8]: Alright, I was wrong: bodytype is supported.
Default faction is also supported.

**Comment by Kenan2000** [comment]
Will upload PRs separately for each mod :-)

**Pull request closed without merging**